services/pull/check.go

```diff
@@ -221,7 +221,7 @@ func getMergeCommit(ctx context.Context, pr *issues_model.PullRequest) (*git.Com
 		RunStdString(&git.RunOpts{Dir: pr.BaseRepo.RepoPath()})
 	if err != nil {
 		return nil, fmt.Errorf("git rev-list --ancestry-path --merges --reverse: %w", err)
-	} else if len(mergeCommit) < git.SHAFullLength {
+	} else if _, err := git.IDFromString(mergeCommit); err == nil {
```

Comment:
It seems the logic changed?

Reply:
You are correct. This check  should be re-arranged and put after the repository is opened so we have proper objectFormat available.

The logic should have been negated `err == nil` to `err != nil` so it's a bad typo here. Fixing...